### PR TITLE
fix(ext/flash) Fix panic when JS caller doesn't consume request body

### LINKED ISSUE #15934: [Flash] Deno has panicked when POST & PUT
File: main.ts
```ts
Deno.serve(() => new Response("Hello, world!"));
```

Request: POST
```
curl --location --request POST 'http://127.0.0.1:9000'
```
Log:
```
RUST_BACKTRACE=full  deno run --allow-net --unstable examples/main.ts
Listening on http://127.0.0.1:9000/

============================================================
Deno has panicked. This is a bug in Deno. Please report this
at https://github.com/denoland/deno/issues/new.
If you can reliably reproduce this panic, include the
reproduction steps and re-run with the RUST_BACKTRACE=1 env
var set and include the backtrace in your report.

Platform: macos x86_64
Version: 1.25.3
Args: ["deno", "run", "--allow-net", "--unstable", "examples/main.ts"]

thread 'main' panicked at 'called `Option::unwrap()` on a `None` value', ext/flash/lib.rs:715:41
stack backtrace:
   0:        0x110256989 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h85521558a183f368
   1:        0x10f7aa31b - core::fmt::write::h01631fae0d2b98bc
   2:        0x110227788 - std::io::Write::write_fmt::h675dde99a2999169
   3:        0x11025b77d - std::panicking::default_hook::{{closure}}::h5e5df492d229fd65
   4:        0x11025b464 - std::panicking::default_hook::h18647b59f1a84ee2
   5:        0x10f71c53b - deno::setup_panic_hook::{{closure}}::hf224392e6471ee9e
   6:        0x11025c2cb - std::panicking::rust_panic_with_hook::hd9ead35a68ccc55e
   7:        0x11025c11f - std::panicking::begin_panic_handler::{{closure}}::h6fca91c5e1dc2f30
   8:        0x11025c0b9 - std::sys_common::backtrace::__rust_end_short_backtrace::h4ff3025d9a0a0490
   9:        0x11025c075 - _rust_begin_unwind
  10:        0x1116080c3 - core::panicking::panic_fmt::h3d9f795ee387ef8d
  11:        0x1116081a7 - core::panicking::panic::h02eb82771823ab30
  12:        0x10fa4ebdf - <core::future::from_generator::GenFuture<T> as core::future::future::Future>::poll::hc074897772ea47be
  13:        0x10fa4daf8 - <extern "C" fn(A0) .> R as v8::support::CFnFrom<F>>::mapping::c_fn::h06031d174ea1ca6c
  14:        0x11069a31e - __ZN2v88internal12_GLOBAL__N_119HandleApiCallHelperILb0EEENS0_11MaybeHandleINS0_6ObjectEEEPNS0_7IsolateENS0_6HandleINS0_10HeapObjectEEENS8_INS0_20FunctionTemplateInfoEEENS8_IS4_EEPmi
  15:        0x1106997f6 - __ZN2v88internal21Builtin_HandleApiCallEiPmPNS0_7IsolateE
```

### PR DESCRIPTION
**Pull request by tsar-boomba**
Fixes #15934, #15759, #15813

If the JS handler gets a POST, PUT, or PATCH request, but doesn't `await` the body, deno would panic because it will try to read the body even though the request has already been handled.

Not sure how/where to test this case, so I could use some help with that.

### CODE REVIEW

File: ext/flash/lib.rs [hunk 1]
```diff
@@ -717,7 +717,14 @@ async fn op_flash_read_body(
...
   };
-  let tx = ctx.requests.get_mut(&token).unwrap();
+  let tx = ctx.requests.get_mut(&token);
+
+  if tx.is_none() {
+    // request was already consumed by caller
+    return 0;
+  }
+
+  let tx = tx.unwrap();
```
Comment (littledivy) [inline ext/flash/lib.rs:727]:
```suggestion
  let tx = match ctx.requests.get_mut(&token) {
    Some(tx) => tx,
    // request was already consumed by caller
    None => return 0,
  };
```

**Comment by bartlomieju** [comment]
@littledivy please take a look

**Comment by ynwd** [comment]
any updates?

**Review by littledivy — APPROVED** [review]
LGTM!

**Pull request merged**